agent/requestor/src/main.rs

```diff
@@ -195,20 +218,32 @@ async fn main() -> anyhow::Result<()> {
         payment_api.clone(),
         started_at,
         agreement_allocation.clone(),
+        if settings.one_agreement {
+            Some(app_abort_handle.clone())
+        } else {
+            None
+        },
```

Comment:
shorter and more-readable -- just a case to know more rust-y way of doing this. You need not to follow me here 
```suggestion
        match settings.one_agreement {
            true => Some(app_abort_handle.clone())
            false => None
        },
```